src/poprox_storage/repositories/experiments.py

```diff
@@ -81,6 +82,47 @@ def store_experiment(
 
         return experiment_id
 
+    def find_recommenders_by_expt_groups(self, date=None):
+        groups_tbl = self.tables["groups"]
+        phases_tbl = self.tables["phases"]
+        recommenders_tbl = self.tables["recommenders"]
+        treatments_tbl = self.tables["treatments"]
+
+        # Find the groups and associated recommenders for the active experiment phases
+        date = date or datetime.date.today()
+
+        group_recommender_query = (
+            select(groups_tbl.c.group_id, recommenders_tbl.c.endpoint_url)
+            .join(treatments_tbl, treatments_tbl.c.group_id == groups_tbl.c.group_id)
+            .join(
+                recommenders_tbl,
+                recommenders_tbl.c.recommender_id == treatments_tbl.c.recommender_id,
+            )
+            .join(phases_tbl, phases_tbl.c.phase_id == treatments_tbl.c.phase_id)
+        ).where(
+            and_(
+                phases_tbl.c.start_date <= date,
+                date < phases_tbl.c.end_date,
+            )
+        )
+
+        result = self.conn.execute(group_recommender_query).fetchall()
+        recommender_lookup_by_group = {row[0]: row[1] for row in result}
+
+        return recommender_lookup_by_group
+
+    def find_accounts_by_expt_groups(self, group_ids):
+        allocations_tbl = self.tables["allocations"]
+
+        # Find accounts allocated to the groups that are assigned the active recommenders above
+        group_query = select(
+            allocations_tbl.c.account_id, allocations_tbl.c.group_id
+        ).where(allocations_tbl.c.group_id.in_(group_ids))
+        result = self.conn.execute(group_query).fetchall()
+        group_lookup_by_account = {row[0]: row[1] for row in result}
+
+        return group_lookup_by_account
```

Comment:
I feel like the return value and function name mis-match a little bit? Reading the function name I would think this returns `Dict[Group, List[Account]]` instead of `Dict[Account, Group]` 

I would have to know how it's planned to be used to decide how I think it would best-work. I can see a world in which `Dict[Group, List[Account]]` is a better return value.

I cannot think of a better name for the function.

Reply:
How about `get_experiment_assignments`? That at least accords better with the return type